src/gui/spatialArrow.js

```diff
@@ -0,0 +1,324 @@
+createNameSpace("realityEditor.gui.spatialArrow");
+
+import * as THREE from '../../thirdPartyCode/three/three.module.js';
+
+(function (exports) {
+    
+    let canvasContainer;
+    let canvas;
+    let ctx;
+    let screenW, screenH;
+    let screenRatio;
+    
+    let color, colorLighter;
+
+    function initService() {
+        console.log('%c spatial arrow init!', 'color: green');
+        addCanvas();
+        resizeCanvas();
+        initCanvas();
+        update();
+    }
+
+    // for testing purpose on js fiddle, don't include it in final code
+    const windowRatio = 1;
+
+    window.addEventListener('resize', () => {
+        resizeCanvas();
+        initCanvas();
+        update();
+    });
+
+    // todo: dynamically set style.top based on menu bar's height -- making the arrow canvas right below the menu bar
+    const menuBarHeight = 30;
```

Comment:
You can use `realityEditor.device.environment.variables.screenTopOffset` to get this dynamically

Reply:
Got it, thx!